doc/source/data/key-concepts.rst

```diff
@@ -146,6 +146,17 @@ scheduling strategy for all Datasets tasks/actors, using the global
   :start-after: __resource_allocation_begin__
   :end-before: __resource_allocation_end__
 
+Fault tolerance
+===============
+
+Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, a ``Dataset`` will be automatically recovered by Ray in case of failures. This works through **lineage reconstruction**: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate these by re-executing the task(s) that created them.
+
+There are a few cases that are not currently supported:
+1. If the original creator of the ``Dataset`` dies. This is because the ``Dataset creator`` stores the metadata for the :ref:`objects <object-fault-tolerance>` that comprise the ``Dataset``.
+2. For a ``DatasetPipeline.split``, we do not support recovery for a consumer failure. When there are multiple consumers, they must all read the split pipeline in lockstep. To recover from this case, the pipeline and all consumers must be restarted together.
```

Comment:
Hm, is this case relevant? My understanding is that DatasetPipeline represents consumption of Datasets and doesn't produce new Datasets (unlike transformations), so the failure in execution DatasetPipeline is not relevant for Dataset FT?

Reply:
I think it is. Even if there is no Dataset transformation involved, DatasetPipeline.split still has to do some computation to split the input dataset.